lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -3455,17 +3458,13 @@ mkApiTransaction timeInterpreter setTimeReference tx = do
     reclaimIfAny :: Natural
     reclaimIfAny
         | tx ^. (#txMeta . #direction) == W.Incoming =
-              if ( totalInWithoutFee > 0 && totalOut > 0 && totalOut > totalInWithoutFee)
-                 && (totalOut - totalInWithoutFee <= depositValue) then
+              if ( totalIn > 0 && totalOut > 0 && totalOut > totalIn)
+                 && (totalOut - totalIn <= depositValue) then
                   depositValue
               else
                   0
         | otherwise = 0
 
-    totalInWithoutFee :: Natural
```

Comment:
indeed, this was redundant if we take into account withdrawals :+1:  And the name was misleading

Reply:
now thinking about this. we want to detect quitting and receiving deposit back. Shouldn't we have instead `totalIn`
```
    totalInWithoutWdrl
        = sum (txOutValue <$> mapMaybe snd (tx ^. #txInputs))
```
in the end?

Reply:
all test passes, so you are right :sweat_smile: